src/consumer.rs

```diff
@@ -574,6 +574,32 @@ impl<T: DeserializeMessage, Exe: Executor> TopicConsumer<T, Exe> {
                         .into());
                     }
                 }
+                Err(ConnectionError::Io(e)) => {
+                    if e.kind() != std::io::ErrorKind::TimedOut {
```

Comment:
To me, you could also retry in the following case as well:

- ConnectionReset
- ConnectionAborted
- NotConnected
- BrokenPipe
- Interrupted
- UnexpectedEof

See [std::io::ErrorKind](https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html)